user-service/src/component-tests/java/uk/gov/gchq/palisade/component/user/service/UserServiceAsyncProxyErrorTest.java

```diff
@@ -0,0 +1,149 @@
+/*
+ * Copyright 2020 Crown Copyright
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.gchq.palisade.component.user.service;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.autoconfigure.cache.CacheAutoConfiguration;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.boot.test.context.SpringBootTest.WebEnvironment;
+import org.springframework.cache.annotation.EnableCaching;
+import org.springframework.core.serializer.support.SerializationFailedException;
+import org.springframework.test.context.ActiveProfiles;
+
+import uk.gov.gchq.palisade.User;
+import uk.gov.gchq.palisade.service.user.config.ApplicationConfiguration;
+import uk.gov.gchq.palisade.service.user.exception.NoSuchUserIdException;
+import uk.gov.gchq.palisade.service.user.model.AuditableUserResponse;
+import uk.gov.gchq.palisade.service.user.model.UserRequest;
+import uk.gov.gchq.palisade.service.user.service.UserServiceAsyncProxy;
+
+import java.util.concurrent.CompletableFuture;
+import java.util.function.Function;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+/**
+ * Tests to verify the handling of exceptions, and the population of audit objects during stream processing
+ */
+@SpringBootTest(
+        classes = {ApplicationConfiguration.class, CacheAutoConfiguration.class},
+        webEnvironment = WebEnvironment.NONE,
+        properties = {"spring.cache.caffeine.spec=expireAfterWrite=1s, maximumSize=100"}
+)
+@EnableCaching
+@ActiveProfiles({"caffeine"})
+class UserServiceAsyncProxyErrorTest {
+
+    private static final String REQUEST_JSON = "{\"userId\":\"test-user-id\",\"resourceId\":\"/test/resourceId\",\"context\":{\"class\":\"uk.gov.gchq.palisade.Context\",\"contents\":{\"purpose\":\"purpose\"}}}";
+    private static final String NO_USER_REQUEST_JSON = "{\"userId\":\"Not-a-real-user\",\"resourceId\":\"/test/resourceId\",\"context\":{\"class\":\"uk.gov.gchq.palisade.Context\",\"contents\":{\"purpose\":\"purpose\"}}}";
```

Comment:
Why have this in a component test? Just create the objects, we don't care about the json

Reply:
Removed all the JSON stuff. Just creates the objects.